src/careamics/models/lvae/noise_models.py

```diff
@@ -134,9 +116,69 @@ def get_noise_model(
     return None
 
 
+class DisentNoiseModel(nn.Module):
```

Comment:
Would be good to get some docs/explanation for this class @ashesh-0

Reply:
Especially likelihood method

Reply:
The purpose of Noise model is to give the likelihood tensor of same dimensions as the prediction. 
For splitting task, we have multiple target channels,and therefore we need multiple noise models. This class is responsible for managing all noise model related stuff. 
self.nmodel_0 is the noise model object for the first class, self.nmodel_1 for the second class and so on. 
likelihood function in this case should return a list of likelihood tensors, one for each channel and using the specific noise_model. 

For HDN, there will be just one noise model and so is returns, for historical reasons, not a list but a single likelihood tensor. `obs.shape[1] ==1`

Reply:
As for the dimensions of the input arguments of the likelihood function, the only constraint is that they should be of the same shape.  For 2D, the shape is `BxCxHxW`, (batch, channel count, height, width). For 3D data, the dimensions will be `BxCxZxHxW`.